content/specific-skill-success-criteria/classes-and-objects/_index.md

```diff
@@ -23,3 +23,122 @@ All students need to provably understand all of the following concepts:
 - getters and setters
 - inheritance and overriding and extending methods
 - composition
+
+# Python
+
+## Constructors
+The `__init__()` method is called the constructor and is always called when an object is created, also known as instantiating an object.
+
+### Default Constructor
+The default constructor is a simple constructor which doesn’t accept any arguments. Its definition has only one argument which is a reference to the instance being constructed.
+```
+class BankAccount:
+  def __init__(self):
+    pass
+```
+
+### Parameterized Constructor
+As the name suggests, a parameterized constructor is a constructor with parameters. Takes its first argument as a reference to the instance being constructed, known as `self` and the rest of the arguments are provided by the programmer.
+```
+class BankAccount:
+  def __init__(self, acc_no, amount):
+    self.acc_no = acc_no
+    self.amount = amount
+```
+
+## Instantiation
+Instantiating a class is creating a copy of the class which inherits all class variables and methods. We simply call the class as if it were a function, passing the arguments that the `__init__()` method defines (except `self`).
+```
+saitama_acc = BankAccount("sa1234", 10000)
+```
+
+A class can have multiple instances.
+```
+kuroko_acc = BankAccount("ka1234", 300)
+tsukasa_acc = BankAccount("ta1234", 75010)
+vash_acc = BankAccount("va1234", 813650)
+```
+
+
+## Getters and setters
+The main purpose that getters and setters serve in object-orientated programs is for ensuring data encapsulation - or put another way, as a mechanism for hiding the implementation details of a class from the user. In Python, getters and setters are not the same as for other object-orientated languages. Private variables in Python are not actually hidden fields. In Python getters and setters are typically used when:
+
+- we want to add validation logic for getting and setting a value
+- to avoid the direct access of a class field so that private variables cannot be accessed directly/modified by an outside user
+
+### Getters
+A getter is a method that gets the value of a property.
+```
+  def balance(self):
+    return self.amount
+```
+
+```
+saitama_acc.amount
+```
+
+```
+saitama_acc.balance()
+```
+
+### Setters
+A setter is a method that sets the value of a property.
+```
+  def deposit(self, amount):
```

Comment:
This isn't a setter at all. This is just a perfectly normal function.

Folks do need to know how instance methods work and how they can change values. But these are not setters, it's the wrong word.